doc/source/reference/c-api/array.rst

```diff
@@ -131,29 +162,6 @@ and its sub-types).
 
     Returns the total number of bytes consumed by the array.
 
-.. c:function:: PyObject *PyArray_BASE(PyArrayObject* arr)
-
-    This returns the base object of the array. In most cases, this
-    means the object which owns the memory the array is pointing at.
-
-    If you are constructing an array using the C API, and specifying
-    your own memory, you should use the function :c:func:`PyArray_SetBaseObject`
-    to set the base to an object which owns the memory.
-
-    If the :c:data:`NPY_ARRAY_WRITEBACKIFCOPY` flag is set, it has a different
-    meaning, namely base is the array into which the current array will
-    be copied upon copy resolution. This overloading of the base property
-    for two functions is likely to change in a future version of NumPy.
-
-.. c:function:: PyArray_Descr *PyArray_DESCR(PyArrayObject* arr)
-
-    Returns a borrowed reference to the dtype property of the array.
-
-.. c:function:: PyArray_Descr *PyArray_DTYPE(PyArrayObject* arr)
-
-    A synonym for PyArray_DESCR, named to be consistent with the
-    'dtype' usage within Python.
-
```

Comment:
I don't really have a strong opinion about keeping these in the same order as the code. I didn't double-check to make sure that the moved docstrings are identical.

Reply:
I understand your point, but since the current C-API documentation relies on handwritten content without Doxygen-generated references, I believe keeping the docstrings in the same order as the code will facilitate easier verification and modifications.